pkg/plugins/backendplugin/instrumentation/instrumentation.go

```diff
@@ -58,6 +65,8 @@ func instrumentPluginRequest(ctx context.Context, cfg Cfg, pluginCtx *backend.Pl
 	pluginRequestDuration.WithLabelValues(pluginCtx.PluginID, endpoint, string(cfg.Target)).Observe(float64(elapsed / time.Millisecond))
 	pluginRequestCounter.WithLabelValues(pluginCtx.PluginID, endpoint, status, string(cfg.Target)).Inc()
 
+	PluginRequestDurationSeconds.WithLabelValues("backend-datasource", pluginCtx.PluginID, endpoint, string(cfg.Target), status).Observe(float64(elapsed / time.Second))
```

Comment:
nice for the `elapsed.Seconds()` !

For the name of the source, I feel like backend/frontend is not enough. Here it is the `Datasource` interface, but it could be multiple things, for both backend and frontend

Reply:
It's not strictly just `Datasource` though because `App`s also are instrumented here too. If you wanted to have `backend_app`/`backend_ds` then you'd need to check based on what settings are set in the `PluginContext`. Having `backend_app|backend_ds|frontend_app|frontend_ds` may be confusing however because a datasource or app can also support backend or frontend implementations...

In an ideal world, since we already have `Target`, `Source` should have a similar structure

```go
type Target string

const (
	TargetNone     Target = "none"
	TargetUnknown  Target = "unknown"
	TargetInMemory Target = "in_memory"
	TargetLocal    Target = "local"
)
```